infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -137,15 +142,41 @@ default OverwriteCause getOverwriteCauseForAddress(long address, LogData entry)
     /**
      * Query if the StreamLog has enough quota to accept writes
      */
-    default boolean quotaExceeded() {
-        return false;
-    }
+    boolean quotaExceeded();
 
     /**
      * Query the exact Quota value in bytes
      * @return the quota set in bytes
      */
-    default long quotaLimitInBytes() {
-        return Long.MAX_VALUE;
-    }
+    long quotaLimitInBytes();
+
+    /**
+     * Query the space used in bytes
+     * @return the space used in bytes
+     */
+    ResourceQuota getLogSizeQuota();
+
+    /**
+     *
+     * @param startAddress the startAddress of a segment file
+     * @return the size of the segment
+     */
+    long getSegmentSize(long startAddress);
+
+    /**
+     * get the log size including startAddress and endAddress
+     * @param startAddress
+     * @param endAddress
+     * @return
+     */
+    default long getSegmentSize(long startAddress, long endAddress) {
+        long  size = 0;
+        startAddress = startAddress/RECORDS_PER_LOG_FILE*RECORDS_PER_LOG_FILE;
```

Comment:
`RECORDS_PER_LOG_FILE` is specific to `StreamLogFiles` and shouldn't be leaked to the interface.

Reply:
+1